sphinx_needs/builder.py

```diff
@@ -157,3 +156,88 @@ def build_needumls_pumls(app: Sphinx, _exception: Exception) -> None:
         needs_builder.set_environment(env)
 
     needs_builder.finish()
+
+
+class NeedsPerPageBuilder(Builder):
+    name = "needs_per_page"
+    format = "needs"
+    file_suffix = ".txt"
+    links_suffix = None
+    LIST_KEY_EXCLUSIONS_NEEDS = ["content_node"]
+
+    def write_doc(self, docname: str, doctree: nodes.document) -> None:
+        pass
+
+    def finish(self) -> None:
+        env = unwrap(self.env)
+        needs = env.needs_all_needs.values()
+        filter_string = self.app.config.needs_builder_filter
+        needs_per_page_build_path = self.app.config.needs_per_page_build_path
+        filtered_needs = filter_needs(self.app, needs, filter_string)
+        needs_per_page_dir = os.path.join(self.outdir, needs_per_page_build_path)
+        needs_per_page_data = {}
+        if not os.path.exists(needs_per_page_dir):
+            os.mkdir(needs_per_page_dir)
+
+        needs_per_page_data_key = []
+        for need in filtered_needs:
```

Comment:
Some more comments would be good, especially if you have such much code in a single function and haven't it broken down to some sub functions.

At least `for`-loops with multiple lines should get a comment, so that you know in advance what the goal of the for loop is before reading all code line.
Something like `sorting needs into docname-based dict` would be helpful.